hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -329,6 +336,21 @@ private ContainerCommandResponseProto dispatchRequest(
     }
   }
 
+  private void updateBCSID(Container container,
+      DispatcherContext dispatcherContext, ContainerProtos.Type cmdType) {
+    long bcsID = container.getBlockCommitSequenceId();
+    long containerId = container.getContainerData().getContainerID();
+    Map<Long, Long> container2BCSIDMap;
+    if (dispatcherContext != null && (cmdType == ContainerProtos.Type.PutBlock
```

Comment:
an alternative implementation would be to ignore the cmdType.
For all requests, read the BCS ID from the map and compare it to the bcsid from the container. if the map value is lower, then update it to the value from the container. 

Advantage:
updateBCSID function becomes de-coupled from the knowledge of which cmdType changes the bcsid.
Disadvantage:
possibly more CPU cost, because every request will pay the cost of reading the bcsid map. but it might be premature to assume that this additional cost will be significant.

Reply:
For all the cmds, the dispatcher context is not setup and will be null. We need to check for specific cmd types to get the context.